### PR TITLE
Fetch Api

### PR DESCRIPTION
**Pull request by danielsafari143**
In this pull request I was able to : 
- Connect to an API within a functional component
- Show different UI outputs depending on the component state

### CODE REVIEW

File: src/components/Quotes.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+
+const Quotes = () => {
+  const [quotes, setQuotes] = useState('Loading...');
```
Comment (VICTOR-CHUKWU) [inline src/components/Quotes.jsx]: - [ ] in managing loading and error, there should be as state to handle them differently, the error state should be false initially while loading state should be true initially

File: src/components/Quotes.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          method: 'GET',
+          headers: { 'X-Api-Key': 'beSNGjWMHbtj5yQK66y8qw==IhBueBkGLQYQ1tJj' },
+          contentType: 'application/json',
+        }).then((res) => res.json())
+        .then((val) => val)
+        .catch((e) => `Error : ${e}`);
+
+      const val = await data;
+      setQuotes(val);
+    };
```
Comment (VICTOR-CHUKWU) [inline src/components/Quotes.jsx]: - [ ] while fetching your API, the loading state should be set to true here and when the API response is successful set to false, if there is an error error state should be set to true also

File: src/components/Quotes.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    getData();
+  }, [setQuotes]);
+
+  if (quotes === 'Loading...') {
+    return <p>{quotes}</p>;
+  }
+
+  if (quotes[0] !== undefined) {
+    return <p className="quote">{quotes[0].quote}</p>;
+  }
```
Comment (VICTOR-CHUKWU) [inline src/components/Quotes.jsx]: - [ ] Here you return loading if the loading  state is true, error if the error state is true and quote if data is fetched successfully

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @danielsafari143  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @danielsafari143,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- [ ] Great job on the funtionality 💯
- [ ] Followed Gitflow 🔥
- [ ] No Linters error ✅
- [ ] All changes are applied🔥🔥

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**